### PR TITLE
[TextField] Fix the spinner buttons behaviour when readOnly

### LINKED ISSUE #1118: TextField readOnly prop ignored when type="number"
# Issue summary

When the `type="number"` prop is present on the TextField component, the `readOnly` prop does not disable or remove the number steppers.  The user can still click the up/down arrows to change the field's value.

## Expected behaviour

Similar to how `disabled` fields are treated when `type="number"`, remove the number steppers to prevent the value from being changed.

## Actual behaviour

The user can click the up/down arrows (the number steppers) to change the field value, despite the `readOnly` prop being present.

## Steps to reproduce the problem

1.  Create a TextField component
1.  Add the `type="number"` prop
1.  Add the `readOnly` prop
1.  Click the up/down arrows (the number steppers) and see the field's value change

## Reduced test case

Check out this [Codesandbox project](https://codesandbox.io/s/p3689vv2m7)

## Specifications

- Are you using the React components? (Y/N): Yes
- Polaris version number: 3.9.0
- Browser: Chrome 72.0.3626.109
- Device: MacBook Pro
- Operating System: macOS Mojave (10.14)

### PR DESCRIPTION
**Pull request by cloneali**
### WHY are these changes introduced?

Resolves #1118 

### WHAT is this pull request doing?
- Add a condition to not render the spinner buttons on TextField when readOnly is 'true' and type is 'number'
- Add a test to confirm the spinner buttons are not rendering when type='number' and readOnly is 'true'

### CODE REVIEW

**Comment by cloneali** [comment]
Opps. I forgot to change the authorship before committing to my personal account. This is causing CLA to failed.

**Comment by tmlayton** [comment]
@cloneali this looks great! I went ahead and reran the CLA check so it passed (after signing it requires a manual rerun until we build the auto-rerun feature 😄)

Can you add a changelog entry to UNRELEASED.md? There is a link in that doc that explains the proper format.

**Comment by cloneali** [comment]
Changelog added to the UNRELEASED.md

File: src/components/TextField/tests/TextField.test.tsx [hunk 1]
```diff
@@ -580,6 +580,19 @@ describe('<TextField />', () => {
...
+      it('removes increment and decrement buttons when readOnly', () => {
+        const element = mountWithAppProvider(
+          <TextField
+            id="MyNumberField"
+            label="NumberField"
+            type="number"
+            readOnly
+          />,
+        );
+        const buttons = element.find('[role="button"]');
```
[the remark this answers is not in the document]
Comment (cloneali) [reply src/components/TextField/tests/TextField.test.tsx]: This is done!

[the remark this answers is not in the document]
Comment (chloerice) [reply src/components/TextField/tests/TextField.test.tsx]: Awesome! Just need to pull the latest down on your local master branch and then go back to this PR's branch and `git rebase master` and it's good to go 🎉

[the remark this answers is not in the document]
Comment (cloneali) [reply src/components/TextField/tests/TextField.test.tsx]: I think its good to go now! 😄

**Pull request merged**